src/WebJobs.Script/SlidingWindow.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class SlidingWindow<TItem>
+    {
+        private readonly TimeSpan _window;
+        private List<Event> _events = new List<Event>();
+        private object _syncLock = new object();
+
+        public SlidingWindow(TimeSpan window)
+        {
+            _window = window;
+        }
+
+        public IEnumerable<TItem> GetEvents(Func<TItem, bool> filter)
+        {
+            lock (_syncLock)
+            {
+                // remove expired events
+                _events.RemoveAll(p => (DateTime.Now - p.TimeStamp) > _window);
```

Comment:
This will be a bit fragile in situations like a clock sync/time change. We should consider changing the event to record the ticks as the timestamp and perform all comparisons using `Ticks` and `UtcNow`. The other alternative is to change the event to use a `StopWatch` instead, but the first option is a bit lighter.

Reply:
Done